src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy };
+        Object.assign(stateCopy, obj.extraData);
+        break;
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
+        obj.keysToRemove.forEach(key => delete stateCopy[key]);
+        break;
+      case 'clear':
+        stateCopy = { };
+        break;
+      default:
+        break;
+    }
+    stateHistory.push(stateCopy);
```

Comment:
you can create a copy when pushing it:
```suggestion
    stateHistory.push({ ...stateCopy });
```

Reply:
✅